index.js

```diff
@@ -111,7 +111,13 @@ class TimeoutError extends Error {
 	}
 }
 
-const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+const delay = ms => new Promise((resolve, reject) => {
+	if (ms > 2147483647) { // See #117, https://stackoverflow.com/questions/3468607/why-does-settimeout-break-for-large-millisecond-delay-values
+		reject(new TypeError('`timeout` can not be greater than 2147483647'));
```

Comment:
`RangeError` seems perfect

Reply:
True, not a type error from js perspective, fixed.

It's a side effect(we control) of `setTimeout`, IMO it's much clearer when it is throw from where the problem lies. Also, this way it handles all possible internal usages of `delay` in the future.

Reply:
To be honest, I expected this discussion. I even wrote the code to walk around this issue so there will be no need to overgeneralize constructor/docs and enforce stronger dependencies but then deleted it because it felt just too stupid to add 6 lines of code because of issue no one is expected to ever hit(now that `false` works). But at this point I can bring it back if you want.

Reply:
I little catch though: won't work for numbers higher than `1e293`, but no one will live long enough to know about that.

Reply:
@sindresorhus I think Got should also throw at this position.